mlflow/langchain/__init__.py

```diff
@@ -0,0 +1,391 @@
+                      `DataType.string` as the column type.
+                      Alternatively, you can explicitly specify the model signature.
+                      The model signature can be :py:func:`inferred
+                      <mlflow.models.infer_signature>` from datasets with valid model input
+                      (e.g. the training dataset with target column omitted) and valid model
+                      output (e.g. model predictions generated on the training dataset),
+                      for example:
+
+                      .. code-block:: python
+
+                        from mlflow.models.signature import infer_signature
+
+                        model = LLMChain(llm=llm, prompt=prompt)
+                        predictions = model.run(input_dataframe)
+                        input_columns = [
+                            {"type": "string", "name": input_key} for input_key in chain.input_keys
+                        ]
+                        signature = infer_signature(input_columns, predictions)
+
+    :param input_example: Input example provides one or several instances of valid
+                          model input. The example can be used as a hint of what data to
+                          feed the model. The given example will be converted to a
+                          Pandas DataFrame and then serialized to json using the
+                          Pandas split-oriented format. Bytes are base64-encoded.
+
+    :param await_registration_for: Number of seconds to wait for the model version
+                        to finish being created and is in ``READY`` status.
+                        By default, the function waits for five minutes.
+                        Specify 0 or None to skip waiting.
+    :param pip_requirements: {{ pip_requirements }}
+    :param extra_pip_requirements: {{ extra_pip_requirements }}
+    :param metadata: Custom metadata dictionary passed to the model and stored in the MLmodel file.
+
+                     .. Note:: Experimental: This parameter may change or be removed in a future
+                                             release without warning.
+    :return: A :py:class:`ModelInfo <mlflow.models.model.ModelInfo>` instance that contains the
+             metadata of the logged model.
+    """
+    import langchain
+
+    if type(lc_model) != langchain.chains.llm.LLMChain:
+        raise TypeError(
+            "MLflow langchain flavor only supports logging langchain.chains.llm.LLMChain "
+            + f"instances, found {type(lc_model)}"
+        )
+    _SUPPORTED_LLMS = {langchain.llms.openai.OpenAI, langchain.llms.huggingface_hub.HuggingFaceHub}
+    if type(lc_model.llm) not in _SUPPORTED_LLMS:
+        logger.warning(
+            "MLflow does not guarantee support for LLMChains outside of HuggingFaceHub and "
+            "OpenAI, found %s",
+            str(type(lc_model.llm)),
+        )
```

Comment:
What might not work for LLMChains outside of HuggingFaceHub and OpenAI?

Reply:
There are many LLMs besides OpenAI and Hugging Face Hub: https://python.langchain.com/en/latest/modules/models/llms/integrations.html